includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -513,9 +513,11 @@ private function maybe_add_lightbox_attributes( $attributes, $node ) {
 
 		$is_file_url                        = preg_match( '/\.\w+$/', wp_parse_url( $parent_node->getAttribute( Attribute::HREF ), PHP_URL_PATH ) );
 		$is_node_wrapped_in_media_file_link = (
-			'a' === $parent_node->tagName
-			&&
-			( 'figure' === $parent_node->tagName || 'figure' === $parent_node->parentNode->tagName )
+			(
+				'a' === $parent_node->tagName
+				||
+				( 'figure' === $parent_node->tagName || 'figure' === $parent_node->parentNode->tagName )
+			)
```

Comment:
Oh wow, yeah. So this fixes a bug. We should add a test for that.

Reply:
Something about the logic in here seems a bit off because in this line here:

https://github.com/ampproject/amp-wp/blob/e311a1089296337738aa3c9e7071eab23f2a4d1b/includes/sanitizers/class-amp-img-sanitizer.php#L514

This would mean effectively that if `$is_file_url` is not empty that `$parent_node` will _always_ be a link because only it would have the `href` attribute (normally). So this line would seem that it would always be true:

https://github.com/ampproject/amp-wp/blob/e311a1089296337738aa3c9e7071eab23f2a4d1b/includes/sanitizers/class-amp-img-sanitizer.php#L517

The whole condition might be better written as:

```php
$is_node_wrapped_in_media_file_link = (
	Tag::A === $parent_node->tagName
	&&
	$parent_node->hasAttribute( Attribute::HREF )
	&&
	// This should be a link to the media file, not the attachment page.
	preg_match( '/\.\w+$/', wp_parse_url( $parent_node->getAttribute( Attribute::HREF ), PHP_URL_PATH ) )
	&&
	( Tag::FIGURE === $parent_node->tagName || Tag::FIGURE === $parent_node->parentNode->tagName )
);

if ( Tag::FIGURE !== $parent_node->tagName && ! $is_node_wrapped_in_media_file_link ) {
	return $attributes;
}
```

But I'm not entirely sure. The conditions necessitating this code are not entirely clear to me.